app/lib/audit/backend.dart

```diff
@@ -26,6 +26,9 @@ AuditBackend get auditBackend => ss.lookup(#_auditBackend) as AuditBackend;
 /// Represents the backend for the audit handling and authentication.
 class AuditBackend {
   final DatastoreDB _db;
+  var _cachedRecords = _CachedRecords(DateTime(0), []);
+  Future<void>? _cacheRecordsUpdateFuture;
```

Comment:
At-least do:
```suggestion
  var _cachedRecords = _CachedRecords(DateTime(0), []);
  Future<void> _cacheRecordsUpdateFuture = Future.value(null);
```
So you never have to handle the null case